subgraphs/opyn-gamma/src/entities/pool.ts

```diff
@@ -0,0 +1,236 @@
+  updateProtocolUSDLocked,
+} from "./protocol";
+
+export function getOrCreatePool(token: Token): LiquidityPool {
+  let pool = LiquidityPool.load(token.id);
+  if (!pool) {
+    pool = new LiquidityPool(token.id);
+    pool.protocol = getOrCreateOpynProtocol().id;
+    pool.name = token.name;
+    pool.symbol = token.symbol;
+    pool.inputTokens = [token.id];
+    pool.outputToken = null;
+    pool.rewardTokens = null;
+    pool.fees = [];
+    pool.oracle = null;
+    pool.createdTimestamp = BIGINT_ZERO;
+    pool.createdBlockNumber = BIGINT_ZERO;
+    pool.totalValueLockedUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeSupplySideRevenueUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeProtocolSideRevenueUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeTotalRevenueUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeEntryPremiumUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeExitPremiumUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeTotalPremiumUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeDepositPremiumUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeWithdrawPremiumUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeTotalLiquidityPremiumUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeVolumeUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeVolumeByTokenAmount = [BIGINT_ZERO];
+    pool.cumulativeVolumeByTokenUSD = [BIGDECIMAL_ZERO];
+    pool.cumulativeDepositedVolumeUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeDepositedVolumeByTokenUSD = [BIGDECIMAL_ZERO];
+    pool.cumulativeDepositedVolumeByTokenAmount = [BIGINT_ZERO];
+    pool.cumulativeWithdrawVolumeUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeWithdrawVolumeUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeWithdrawVolumeByTokenUSD = [BIGDECIMAL_ZERO];
+    pool.cumulativeWithdrawVolumeByTokenAmount = [BIGINT_ZERO];
+    pool.cumulativeExercisedVolumeUSD = BIGDECIMAL_ZERO;
+    pool.cumulativeClosedVolumeUSD = BIGDECIMAL_ZERO;
+    pool.openInterestUSD = BIGDECIMAL_ZERO;
+    pool.putsMintedCount = INT_ZERO;
+    pool.callsMintedCount = INT_ZERO;
+    pool.contractsMintedCount = INT_ZERO;
+    pool.contractsTakenCount = INT_ZERO;
+    pool.contractsExpiredCount = INT_ZERO;
+    pool.contractsExercisedCount = INT_ZERO;
+    pool.contractsClosedCount = INT_ZERO;
+    pool.openPositionCount = INT_ZERO;
+    pool.closedPositionCount = INT_ZERO;
+    pool.inputTokenBalances = [BIGINT_ZERO];
+    pool.inputTokenWeights = [BIGDECIMAL_ONE];
+    pool.outputTokenSupply = null;
+    pool.outputTokenPriceUSD = null;
+    pool.stakedOutputTokenAmount = null;
+    pool.rewardTokenEmissionsAmount = null;
+    pool.rewardTokenEmissionsUSD = null;
+    pool._lastDailySnapshotTimestamp = BIGINT_ZERO;
+    pool._lastHourlySnapshotTimestamp = BIGINT_ZERO;
+    pool.save();
+    incrementProtocolTotalPoolCount();
+  }
+  return pool;
+}
+
+export function handlePoolDeposit(
+  event: ethereum.Event,
+  pool: LiquidityPool,
+  deposit: Deposit
+): void {
+  const amount = deposit.inputTokenAmounts[0];
+  pool.inputTokenBalances = [pool.inputTokenBalances[0].plus(amount)];
```

Comment:
For inputTokenBalance would it be more accurate to use this contract call (https://etherscan.io/address/0x5934807cC0654d46755eBd2848840b616256C6Ef#readContract#F3) ? There is a difference in the values from what I see

Reply:
Hmmm, maybe I was misreading something. It looks good to me! Sorry for any confusion